api/grpcserver/activation_service.go

```diff
@@ -53,3 +56,26 @@ func (s *activationService) Get(ctx context.Context, request *pb.GetRequest) (*p
 		Atx: convertActivation(atx),
 	}, nil
 }
+
+func (s *activationService) Highest(ctx context.Context, req *empty.Empty) (*pb.HighestResponse, error) {
+	highest, err := s.atxProvider.MaxHeightAtx()
+	if err != nil {
+		return &pb.HighestResponse{
+			Atx: &pb.Activation{
+				Id: &pb.ActivationId{Id: s.goldenAtx.Bytes()},
+			},
+		}, nil
+	}
+	atx, err := s.atxProvider.GetFullAtx(highest)
+	if err != nil || atx == nil {
+		return nil, status.Error(codes.NotFound, fmt.Sprintf("atx id %v not found: %v", highest, err.Error()))
+	}
+
+	if highest != atx.ID() {
```

Comment:
I think this is probably not needed. If that check fails that would mean that `GetFullAtx(atxID)` returned an ATX that doesn't have `atxID` as its ID.

Reply:
i don't fully understand. it's unlikely but better not return something that's wrong in API.
why do you think it's not needed?

Reply:
i think as well that this check looks strange, it is a huge programmer mistake if it happens.
there should be a way not to write things like this

Reply:
ok. in that case do you object that i remove the same check in Get()?

Reply:
I wasn't aware that there was such a check, but yes, imo it can go there as well 🙂